swarm/src/lib.rs

```diff
@@ -475,15 +475,8 @@ where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
         &mut self.behaviour
     }
 
-    /// Returns the next event that happens in the `Swarm`.
-    ///
-    /// Includes events from the `NetworkBehaviour` but also events about the connections status.
-    pub async fn next_event(&mut self) -> SwarmEvent<TBehaviour::OutEvent, THandleErr> {
-        future::poll_fn(move |cx| ExpandedSwarm::poll_next_event(Pin::new(self), cx)).await
-    }
-
     /// Returns the next event produced by the [`NetworkBehaviour`].
```

Comment:
```suggestion
    /// Convenience function only returning events produced by the [`NetworkBehaviour`].
    ///
    /// For all events via [`SwarmEvent`] see [`Stream`] implementation on [`ExpandedSwarm`].
```

To stress the fact that this is just a filter on top of the `Stream` implementation. Might help newcomers. What do you think @elenaf9?

Reply:
> Long story short, what do you think of offering both `poll_next_behaviour_event` and `next_behaviour_event` 

I agree that the `Poll` style should be provided, but wouldn't `Stream::poll_next_unpin()` (and matching on `SwarmEvent::Behaviour`) also implement the necessary logic? I think (please correct me if I am wrong) this is more a question of convenience vs complexity, and how I understood you initial comment on #1876, the goal is to reduce the complexity. Or do you think it is better to add it anyways @mxinden?